src/transforms/sample.jl

```diff
@@ -18,50 +19,41 @@ Sample(1_000, replace=false, ordered=true)
 # with rng
 using Random
 rng = MersenneTwister(2)
-Sample(rng, 1_000)
+Sample(1_000, rng=rng)
 
 # with weights
-using StatsBase
-wv = pweights([0.2, 0.1, 0.3])
-Sample(wv, 1_000)
-Sample(rng, wv, 1_000)
+Sample(10, rand(100))
 ```
 """
-struct Sample{R<:AbstractRNG,W} <: Stateless
-  rng::R
-  wv::W
-  n::Int
+struct Sample{W,RNG} <: Stateless
+  size::Int
+  weights::W
   replace::Bool
   ordered::Bool
+  rng::RNG
 end
 
-Sample(rng::AbstractRNG, wv::AbstractWeights, n::Int; replace=true, ordered=false) =
-  Sample(rng, wv, n, replace, ordered)
-
-Sample(rng::AbstractRNG, n::Int; replace=true, ordered=false) =
-  Sample(rng, nothing, n, replace, ordered)
-
-Sample(wv::AbstractWeights, n::Int; replace=true, ordered=false) =
-  Sample(GLOBAL_RNG, wv, n, replace, ordered)
-
-Sample(n::Int; replace=true, ordered=false) =
-  Sample(GLOBAL_RNG, nothing, n, replace, ordered)
+Sample(size::Int, weights=[];
+       replace=false, ordered=false,
+       rng=Random.GLOBAL_RNG) =
+  Sample(size, weights, replace, ordered, rng)
```

Comment:
```suggestion
Sample(size::Int; replace=false, ordered=false, rng=Random.GLOBAL_RNG) =
  Sample(size, nothing, replace, ordered, rng)
  
Sample(size::Int, weights::AbstractWeights;
       replace=false, ordered=false,
       rng=Random.GLOBAL_RNG) =
  Sample(size, weights, replace, ordered, rng)
  
Sample(size::Int, weights; kwargs...) = Sample(size, Weights(collect(weights)); kwargs...)
```

I think this solution is more generic.

Reply:
Our main purpose in this PR is to eliminate this variation of AbstractWeights types that StatsBase.jl introduces. In this use case where we want to sample rows from a table with weights, we just mean probability weights, or weights that are proportional to the inverse frequency.